src/storage/accessors/InMemoryDataAccessor.ts

```diff
@@ -0,0 +1,155 @@
+  }
+
+  public async getData(identifier: ResourceIdentifier): Promise<Readable> {
+    const entry = this.getEntry(identifier);
+    if (!this.isDataEntry(entry)) {
+      throw new NotFoundHttpError();
+    }
+    return this.copyData(entry);
+  }
+
+  public async getMetadata(identifier: ResourceIdentifier): Promise<RepresentationMetadata> {
+    const entry = this.getEntry(identifier);
+    if (this.isDataEntry(entry) === identifier.path.endsWith('/')) {
+      throw new NotFoundHttpError();
+    }
+    return this.generateMetadata(identifier, entry);
+  }
+
+  public async getNormalizedMetadata(identifier: ResourceIdentifier): Promise<RepresentationMetadata> {
+    const entry = this.getEntry(identifier);
+    return this.generateMetadata(identifier, entry);
+  }
+
+  public async writeDataResource(identifier: ResourceIdentifier, data: Readable, metadata?: RepresentationMetadata):
+  Promise<void> {
+    const { parent, name } = this.getParentEntry(identifier);
+    parent.entries[name] = {
+      // Drain original stream and create copy
+      data: streamifyArray(await arrayifyStream(data)),
+      metadata,
+    };
+  }
+
+  public async writeContainer(identifier: ResourceIdentifier, metadata?: RepresentationMetadata): Promise<void> {
+    const { parent, name } = this.getParentEntry(identifier);
+    parent.entries[name] = {
+      entries: {},
+      metadata,
+    };
+  }
+
+  public async modifyResource(): Promise<void> {
+    throw new Error('Not supported.');
+  }
+
+  public async deleteResource(identifier: ResourceIdentifier): Promise<void> {
+    const { parent, name } = this.getParentEntry(identifier);
+    if (!parent.entries[name]) {
+      throw new NotFoundHttpError();
+    }
+    // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+    delete parent.entries[name];
+  }
+
+  private isDataEntry(entry: CacheEntry): entry is DataEntry {
+    return Boolean((entry as DataEntry).data);
+  }
+
+  private getParentEntry(identifier: ResourceIdentifier): { parent: ContainerEntry; name: string } {
+    let parts = identifier.path.slice(this.base.length).split('/').filter((part): boolean => part.length > 0);
+
+    // Workaround for when identifier is root
+    if (parts.length === 0) {
+      return { parent: { entries: { root: this.store }}, name: 'root' };
+    }
+
+    const name = parts.slice(-1)[0];
+    parts = parts.slice(0, -1);
+    let parent = this.store;
+
+    for (const part of parts) {
+      const child = parent.entries[part];
+      if (!child || this.isDataEntry(child)) {
+        throw new NotFoundHttpError();
+      }
+      parent = child;
+    }
+
+    return { parent, name };
+  }
+
+  private getEntry(identifier: ResourceIdentifier): CacheEntry {
+    const { parent, name } = this.getParentEntry(identifier);
+    const entry = parent.entries[name];
+    if (!entry) {
+      throw new NotFoundHttpError();
+    }
+    return entry;
+  }
+
+  private copyData(source: DataEntry): Readable {
```

Comment:
Isn't there a better method? The `highWaterMark` thing was just a hack.

Reply:
True, I should drain the stream into an array.